e107_plugins/pm/pm.php

```diff
@@ -373,17 +388,21 @@ function show_pm($pmid, $comeFrom = '')
 	public function showBlocked($start = 0)
 	{
 		$tp = e107::getParser();
-		$tpl_file = THEME.'pm_template.php';
-		include(is_readable($tpl_file) ? $tpl_file : e_PLUGIN.'pm/pm_template.php');
+//		$tpl_file = THEME.'pm_template.php';
+//		include(is_readable($tpl_file) ? $tpl_file : e_PLUGIN.'pm/pm_template.php');
+    $PM_BLOCKED = e107::getTemplate('pm', 'pm', 'blocked');
+
 		$pmBlocks = $this->block_get_user();			// TODO - handle pagination, maybe (is it likely to be necessary?)
 
 		$sc = e107::getScBatch('pm',TRUE, 'pm');
 		$sc->pmBlocks = $pmBlocks; 
+  $sc->wrapper('pm');
 	
-		$PM_BLOCKED_HEADER = $this->updateTemplate($PM_BLOCKED_HEADER);
-		$PM_BLOCKED_TABLE  = $this->updateTemplate($PM_BLOCKED_TABLE);
-		$PM_BLOCKED_EMPTY  = $this->updateTemplate($PM_BLOCKED_EMPTY);
-		$PM_BLOCKED_FOOTER  = $this->updateTemplate($PM_BLOCKED_FOOTER);
+// Is updateTemplate really necessary for v2.x templates?
+		$PM_BLOCKED_HEADER = $this->updateTemplate($PM_BLOCKED['start']);
+		$PM_BLOCKED_TABLE  = $this->updateTemplate($PM_BLOCKED['item']);
+		$PM_BLOCKED_EMPTY  = $this->updateTemplate($PM_BLOCKED['empty']);
+		$PM_BLOCKED_FOOTER  = $this->updateTemplate($PM_BLOCKED['end']);
 	
```

Comment:
I know, i just saw it on forum plugin.
But THEME_LEGACY is dependen of the existance of theme.xml.
Imagine you got a old theme template "injected" in the templates folder of a v2.x theme.
What happens? Caos....

Anyway, if the problems is the handling of legacy pm plugin templates, i can add the check, but it will lead to the replacement of gettemplate from the plugin, [since the gettemplate only checks for the templates in 2 places](https://github.com/e107inc/e107/blob/ca1fd4c40c8c34500f9655833f9dd7107a0c5d15/e107_handlers/e107_class.php#L2682):
Example usage: <code>e107::getTemplate('user', 'short_start');</code>
	 * - e107_themes/current_frontend_theme/templates/user_template.php (if $override is true)
	 * - e107_themes/templates/user_template.php (if override not found or $override is false)

So, what's the point on full upgrading to V2.x? If it's to deal with legacy templates, might better leave it this way...

As i stated before, if a plugin is V2, it should be FULLY V2, not just parts of the code.
If there's core code that can't handle legacy, that's another issue....